src/Nethermind/Nethermind.Optimism/OptimismTransactionProcessor.cs

```diff
@@ -0,0 +1,203 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.State;
+
+namespace Nethermind.Optimism;
+
+public class OptimismTransactionProcessor : TransactionProcessor
+{
+    private readonly IL1CostHelper _l1CostHelper;
+    private readonly IOPConfigHelper _opConfigHelper;
+
+    public OptimismTransactionProcessor(
+        ISpecProvider? specProvider,
+        IWorldState? worldState,
+        IVirtualMachine? virtualMachine,
+        ILogManager? logManager,
+        IL1CostHelper l1CostHelper,
+        IOPConfigHelper opConfigHelper) : base(specProvider, worldState, virtualMachine, logManager)
+    {
+        _l1CostHelper = l1CostHelper;
+        _opConfigHelper = opConfigHelper;
+    }
+
+    protected override void Execute(Transaction tx, BlockExecutionContext blCtx, ITxTracer tracer, ExecutionOptions opts)
+    {
+        if (tx.IsDeposit())
+        {
+            IReleaseSpec spec = SpecProvider.GetSpec(blCtx.Header);
+
+            WorldState.AddToBalanceAndCreateIfNotExists(tx.SenderAddress!, tx.Mint, spec);
+
+            if (opts.HasFlag(ExecutionOptions.Commit) || !spec.IsEip658Enabled)
+                WorldState.Commit(spec);
+        }
+
+        base.Execute(tx, blCtx, tracer, opts);
+    }
+
+    protected override bool ValidateStatic(Transaction tx, BlockHeader header, IReleaseSpec spec, ITxTracer tracer, ExecutionOptions opts,
+        out long intrinsicGas)
+    {
+        bool result = base.ValidateStatic(tx, header, spec, tracer, opts, out intrinsicGas);
+        if (tx.IsDeposit() && !tx.IsOPSystemTransaction && !result)
+        {
+            if (!WorldState.AccountExists(tx.SenderAddress!))
+            {
+                WorldState.CreateAccount(tx.SenderAddress!, 0, 1);
+            }
+            else
+            {
+                WorldState.IncrementNonce(tx.SenderAddress!);
+            }
```

Comment:
A bit of hack that `Validate` mutates the state

Reply:
I would love to push it to the next pr. Will require refactoring

Reply:
let's also capture this in an issue

Reply:
#6221